### PR TITLE
Refactored logic in RankIcon and CharIcon Components

### PR DESCRIPTION
**Pull request by cityscapedview**
**What did you do?**
Created SearchImg function and called it inside of CharIcon and RankIcon components.  

**Why did you do it?**
Refactoring of the duplicate logic in two components is compliant with DRY programming standards.

**Screenshot before:**
No UI change, view code for logical refactoring.

**Screenshot after:**
No UI change, view code for logical refactoring.

### CODE REVIEW

File: ComposureWatch/src/Components/SearchImg.jsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const SearchImg = (attribute, list) => {
```
Comment (zacharyrankin) [inline ComposureWatch/src/Components/SearchImg.jsx:1]: In this case, the `SearchImg` function isn't really a component; it is more of a utility function, so I recommend renaming it to be lowercase, e.g., searchImage or getImage, and putting it into a `utils.js` file outside of the Components directory. (If you decide to keep it that is)

Normally the `UpperCase` convention is applied to classes or components.

File: ComposureWatch/src/Components/RankIcon.jsx [hunk 2]
```diff
@@ -38,11 +39,7 @@ const RankIcon = ({ rank }) => {
     },
   ];
 
-  let charRank;
-  let result = rankList.find((item) => item.id === rank);
-  if (result) {
-    charRank = result.img;
-  }
+  let charRank = SearchImg(rank, rankList);
```
Comment (zacharyrankin) [inline ComposureWatch/src/Components/RankIcon.jsx:42]: Do you want to do any special handling the rank passed is not found? Maybe just a placeholder image or something?

File: ComposureWatch/src/Components/RankIcon.jsx [hunk 3]
```diff
@@ -38,11 +39,7 @@ const RankIcon = ({ rank }) => {
     },
   ];
 
-  let charRank;
-  let result = rankList.find((item) => item.id === rank);
-  if (result) {
-    charRank = result.img;
-  }
```
Comment (zacharyrankin) [inline ComposureWatch/src/Components/RankIcon.jsx:45]: If you did want to shorten up this code you could do something like:
```
const rankImg = rankList.find(({ id }) => id === rank)?.img;
```

this uses destructuring and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

File: ComposureWatch/src/Components/CharIcon.jsx [hunk 4]
```diff
@@ -183,11 +184,7 @@ const CharIcon = ({ character }) => {
     },
   ];
 
-  let charImg;
-  let result = charList.find((item) => item.id === character);
-  if (result) {
-    charImg = result.img;
-  }
+  let charImg = SearchImg(character, charList);
```
[the remark this answers is not in the document]
Comment (cityscapedview) [reply ComposureWatch/src/Components/CharIcon.jsx:187]: Those articles were good reads and are applicable to this situation.  I think you are right about wanting to include ranking order in the rank component.  I am going to leave the code how it was.

**Pull request closed without merging**